README.md

```diff
@@ -1,2 +1,106 @@
+
+Furthermore, the CI server saves the build history, which persists even if the server is rebooted.
+
+## Overview
+The CI server is a Maven project and contains three top folders inside the DD2480-CI project folder:
+* **src**: contains the source code of the server.
+  * **main**:
+    * **BuildHistoryManager**:Manages the build history of a continuous integration server
+    * **BuildStatus**: An enum containing the different types of commit statuses
+    * **ContinuousIntegrationServer**: Houses the server, based on the skeleton found [here](https://github.com/KTH-DD2480/smallest-java-ci).
+    * **GitRepo**: Clones the repo to the server
+    * **GitStatusUpdate**: Sends a POST request to the GitHub REST API to update a commit’s status
+    * **MavenInvokerBuilder**: Executes the build for Maven repository and retrieves the build output and result status
+    * **RequestParser**: Checks if an HTTP request represents a GitHub push event and extracts information needed for build and notification.
+    * **TestAutomationHandler**: Executing the automated tests of the commit on the branch where the change has been made
+      and sends notification to GitHub
+  * test: contains all tests for the created classes
+
+
+## Build requirements
+* JDK 21
+* GitHub account
+* Personal Access Token (PAT)
+
+### Build instructions
+To build the project, go into the root directory of the project and run `mvn test` in your terminal. This will compile the project and run all tests using JUnit.
+
+### Webhook
+Our server exposes the port `8012`, and we recommend using `ngrok` to tunnel incoming requests. The server triggers the build using webhook requests from GitHub.
+
+### Configuring a Personal Access Token 
+This server sends requests to the GitHub REST API, which requires a Personal Access Token.
+
+Create a file called `.env` in the root of your project, then follow these [steps](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens) to generate a personal access token (give it `repo` access). Then, put the line `ci_token=<YOUR TOKEN HERE>` into the .env file.
+
+## Running the server ##
+1. Set up port forwarding and Webhook for your repo
+2. run `mvn test` in the root of your project to trigger the building process locally
+3. run `mvn clean compile exec:java` to start the server
+4. Go to `localhost:8012` in your browser
+5. Push a commit to your repo
+
+## Features
+
+### Compilation 
+...
+
+### Cloning the repo
+Our server will automatically clone and checkout the correct commit in the repo using the method cloneRepo and checkoutCommit, implemented with Jgit in the GitRepo.java file. The cloned repo will be stored in a temporary folder which will be deleted once the CI server is finished.
+
+### Execution of automated tests
```

Comment:
Our server will analyze the HTTP request received from the Webhook, extract the branch, clone URL, and commit IDs. These details will be used to clone the git repository and add build history. The compilation process will then test the repository and update the relevant notification and history log accordingly.